### PR TITLE
Add xmlsec

### PR DESCRIPTION
**Pull request by johnfosborneiii**
#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates

### CODE REVIEW

**Comment by dlorenc** [comment]
I think you have to disable the update check here - we don't really support the mangled-package-version stuff in that direction, it's getting confused. cc @rawlingsj

File: xmlsec.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      runtime:
+    description: xmlsec dev
+
+  - name: xmlsec-doc
+    pipeline:
+      - uses: split/manpages
+    description: xmlsec manpages
+
+update:
+  enabled: true
```
Comment (rawlingsj) [inline xmlsec.yaml:94]:
```suggestion
  enabled: true
  version-separator: _
```
Adding this should work.  I just did a quick check using
```
wolfictl check update  --override-version 0.0.1 xmlsec.yaml
```

Comment (rawlingsj) [reply xmlsec.yaml:94]: I did notice that there's a new `version-transform` config available [example](https://github.com/wolfi-dev/os/blob/0ca90c76b501d45c35e13e74049a54c18c39bfb3/openjdk-20.yaml#L221C1-L223C19),  in the update config but I don't think it's wired up correctly yet. I can help do that early next week unless it's already being looked at.  Once that does work we can switch all places that use `version-separator` config to `version-transform`.  

For now the suggestion above should make CI green and mean auto updates work.

Comment (kaniini) [reply xmlsec.yaml:94]: `version-transform` should be working fine.  What issues are you seeing with it?

**Review by kaniini — APPROVED** [review]

**Comment by kaniini** [comment]
> I think you have to disable the update check here - we don't really support the mangled-package-version stuff in that direction, it's getting confused. cc @rawlingsj

We do, with `version-transform`

**Pull request merged**